kmk/common/internal_keycodes.py

```diff
@@ -16,13 +16,54 @@ def process_internal_key_event(state, action, changed_key, logger=None):
         return tg(state, action, changed_key, logger=logger)
     elif changed_key.code == Keycodes.Layers._KC_TO:
         return to(state, action, changed_key, logger=logger)
+    elif changed_key.code == Keycodes.KMK.KC_GESC.code:
+        return grave_escape(action, state, logger=logger)
     else:
         return state
 
 
-def tilde(state, action, changed_key, logger):
-    # TODO Actually process keycodes
-    return state
+def grave_escape(action, state, logger):
+    if action['type'] == KEY_DOWN_EVENT:
+        for key in state.keys_pressed:
+            if (key == Keycodes.Modifiers.KC_LSHIFT or
```

Comment:
This and the `elif` block can be condensed into a single `set` object with an `in` check, but this technically still works